lib/utils/getDeepestItemDepth.js

```diff
@@ -1,17 +1,23 @@
 // @flow
-import { Document, Node } from 'slate';
+import { Editor, type Path } from 'slate';
+import { type Options } from '..';
+import { isItem } from '.';
 
 /**
  * Find all `list_item` descendants of a node and retrieve the deepest depth.
  */
-function getDeepestItemDepth(node: Node, document: Document) {
-    return node
-        .filterDescendants(descendant => descendant.type === 'list_item')
-        .reduce(
-            (maxLevel, descendant) =>
-                Math.max(maxLevel, document.getDepth(descendant.key)),
-            0
-        );
-}
 
-export default getDeepestItemDepth;
+export const getDeepestItemDepth = (options: Options) => (
+    editor: Editor,
+    path: Path
+): number =>
+    Array.from(
+        Editor.nodes(editor, {
+            at: path,
+            match: isItem(options),
+        })
+    ).reduce(
+        (maxLevel, [, itemPath]) =>
+            Math.max(maxLevel, itemPath.length - path.length),
+        0
+    );
```

Comment:
Could we stick with using generator and go with `for .. of`

Reply:
For some reason our ESLint config disallows the use of for of... I at least replaced `Array.from` with array destructuring.